File: src/agents.ts
import { MODELS, ProviderName } from './config';
import * as P from './prompts';

export interface AgentDef {
  name: string;
  role: string;
  provider: ProviderName;
  model: string;
  system: string;
  /** Display color (ANSI 256) for the terminal transcript. */
  color: number;
}

// Provider assignment is deliberate: Advocate (Claude) debates Opposing (GPT), and
// Adjudicator A (Claude) is checked against Adjudicator B (Gemini) — different model
// families, so the consensus check is genuinely independent.
export const AGENTS = {
  intake: {
    name: 'Intake Parser',
    role: 'Extract the incident facts from the claim',
    provider: 'openai',
    model: MODELS.intake,
    system: P.INTAKE_PROMPT,
    color: 245,
  },
  evidence: {
    name: 'Evidence Aggregator',
    role: 'Build the grounded Evidence Ledger',
    provider: 'gemini',
    model: MODELS.evidence,
    system: P.EVIDENCE_PROMPT,
    color: 109,
  },
  advocate: {
    name: 'Liability Advocate',
    role: 'Argue our insured is owed recovery',
    provider: 'anthropic',
    model: MODELS.advocate,
    system: P.ADVOCATE_PROMPT,
    color: 39,
  },
  opposing: {
    name: 'Opposing-Carrier Red Team',
    role: 'Attack our case like the other insurer',
    provider: 'openai',
    model: MODELS.opposing,
    system: P.OPPOSING_PROMPT,
    color: 203,
  },
  adjudicator: {
    name: 'Adjudicator A',
    role: 'Neutrally set fault % and recovery (Claude)',
    provider: 'anthropic',
    model: MODELS.adjudicator,
    system: P.ADJUDICATOR_PROMPT,
    color: 178,
  },
  adjudicator_b: {
    name: 'Adjudicator B',
    role: 'Independent adjudicator on a different family (Gemini)',
    provider: 'gemini',
    model: MODELS.adjudicator_b,
    system: P.ADJUDICATOR_PROMPT,
    color: 214,
  },
  verifier: {
    name: 'Source-Alignment Verifier',
    role: 'Audit every cited claim against its source fact',
    provider: 'gemini',
    model: MODELS.verifier,
    system: P.VERIFIER_PROMPT,
    color: 105,
  },
  drafter: {
    name: 'Demand Letter Drafter',
    role: 'Write the formal demand letter',
    provider: 'anthropic',
    model: MODELS.drafter,
    system: P.DRAFTER_PROMPT,
    color: 141,
  },
} satisfies Record<string, AgentDef>;

export type AgentKey = keyof typeof AGENTS;

File: backend/app/agents.py
"""Agent definitions (ported from src/agents.ts). Each agent maps to a Band agent
identity (agent_id + api_key) loaded from band_config when running on real Band.
"""
from __future__ import annotations
from dataclasses import dataclass

from . import prompts as P
from .config import MODELS


@dataclass(frozen=True)
class AgentDef:
    key: str
    name: str
    role: str
    provider: str
    model: str
    system: str
    color: int  # ANSI 256 for the CLI; the web UI maps by name


# Provider assignment is deliberate: the Advocate (Claude) debates the Opposing
# red team (GPT), and Adjudicator A (Claude) is checked against Adjudicator B
# (Gemini) — different model families, so the consensus check is truly independent.
AGENTS: dict[str, AgentDef] = {
    "intake": AgentDef("intake", "Intake Parser", "Extract the incident facts from the claim", "openai", MODELS["intake"], P.INTAKE_PROMPT, 245),
    "evidence": AgentDef("evidence", "Evidence Aggregator", "Build the grounded Evidence Ledger", "gemini", MODELS["evidence"], P.EVIDENCE_PROMPT, 109),
    "advocate": AgentDef("advocate", "Liability Advocate", "Argue our insured is owed recovery", "anthropic", MODELS["advocate"], P.ADVOCATE_PROMPT, 39),
    "opposing": AgentDef("opposing", "Opposing-Carrier Red Team", "Attack our case like the other insurer", "openai", MODELS["opposing"], P.OPPOSING_PROMPT, 203),
    "adjudicator": AgentDef("adjudicator", "Adjudicator A", "Neutrally set fault % and recovery (Claude)", "anthropic", MODELS["adjudicator"], P.ADJUDICATOR_PROMPT, 178),
    "adjudicator_b": AgentDef("adjudicator_b", "Adjudicator B", "Independent adjudicator on a different family (Gemini)", "gemini", MODELS["adjudicator_b"], P.ADJUDICATOR_PROMPT, 214),
    "verifier": AgentDef("verifier", "Source-Alignment Verifier", "Audit every cited claim against its source fact", "gemini", MODELS["verifier"], P.VERIFIER_PROMPT, 105),
    "drafter": AgentDef("drafter", "Demand Letter Drafter", "Write the formal demand letter", "anthropic", MODELS["drafter"], P.DRAFTER_PROMPT, 141),
}
